Modified the Auth pages

sanketbansal opened the pull request:
<!--
(Thanks for sending a pull request! .)
-->

<!--The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: Introduce the first version of the collection editor." -->

<!-- Add issue number here. If you do not solve the issue entirely, please change the message e.g. "Addresses #IssueNumber -->


#### Changes proposed in this pull request:

<!-- Changes: Add here what changes were made in this issue and if possible provide links. -->
- Modified login and signup components to make UI similar to current evalAI UI.

<!-- Demo Link: Add here the link where you changes can be seen. -->
- Link to live demo: http://pr-139-evalai.surge.sh <!-- Replace XXX with your PR no: -->

<!-- Screenshots for the change: Add here the screenshot of the fix. -->
-
![Screenshot from 2019-06-05 22-48-40](https://user-images.githubusercontent.com/17106489/59004895-58e45700-87e9-11e9-9173-b2fa020345d6.png)
![Screenshot from 2019-06-05 22-49-05](https://user-images.githubusercontent.com/17106489/59004905-600b6500-87e9-11e9-9cf4-a1296743828b.png)

sanketbansal committed:
Ported image assets

Ported the image assets from angularJs application into 'image-port' directory

sanketbansal committed:
Added styles from angularjs

Ported styles from angularjs Application inside style-port folder

sanketbansal committed:
Modified auth service

Added isAuth parameter in auth service

sanketbansal committed:
Modified auth service

Changes made in auth services to integrate angularJS authentication features

sanketbansal committed:
Changes in app component

Modified app component to show loader according to authService 'isLoader' variable.

sanketbansal committed:
Modified app loader component

app loader component modified to accompany AngularJS application UI.

sanketbansal committed:
Modified angular.json file

Added main.scss file of ported style in the angular.json file to compile it at the time of building project.

sanketbansal committed:
Modified auth component

Changes made in auth component to acquire angularJS application UI.

sanketbansal committed:
Modified Login component

Added changes in login component to make it functional like angularJS application.

sanketbansal committed:
Modified signup component

Modified the functionality of signup component to make it work like AngularJS Application.

sanketbansal committed:
Modified verify-email component

Changes made in the verify-email component functionality

sanketbansal committed:
Added password mismatch validator directive

password mismatch validator directive added to check the mismatch in password & confirm password during signup.

sanketbansal committed:
Removed routing from auth service

Routing to 'auth/login' has been removed

sanketbansal committed:
Make auth instance public

Auth service instance in auth and signup component made public.

sanketbansal committed:
Added form module in test files

Form module added in test file to provide ngModel and ngForm directive.

lunayach commented:
![auth](https://user-images.githubusercontent.com/21227893/59006468-afd34600-8854-11e9-8ea8-619f6fed2005.png)

@sanketbansal there seems to be a couple of issues with the UI.

- Sign Up button overlaying the "Passwords do not match message".
- When I am writing in the text fields, placeholders are still there.

lunayach commented on src/app/components/auth/login/login.component.html:
```diff
@@ -1,12 +1,55 @@
+  <ul class="left">
+    <li class="ev-horiz-list"><a routerLink="/auth/login" class="dark-link" routerLinkActive="active-auth" (click)="authService.resetForm()">Log In</a></li>
+    <li class="ev-horiz-list"><a routerLink="/auth/signup" class="light-link" routerLinkActive="active-auth" (click)="authService.resetForm()">Sign Up</a></li>
+    <!-- <li class="ev-horiz-list"><a ui-sref="auth.reset-password" class="light-link" ui-sref-active="active-auth" ng-click="auth.resetForm()">Forgot Password</a></li> -->
```
Commented code.

lunayach replied:
Resolve this. :slightly_smiling_face:

lunayach commented on src/app/components/auth/login/login.component.ts:
```diff
@@ -85,31 +59,56 @@ export class LoginComponent implements OnInit, AfterViewInit {
     let redirectTo = '/dashboard';
     const REDIRECT_URL = self.globalService.getData(self.globalService.redirectStorageKey);
     if (REDIRECT_URL && REDIRECT_URL['path']) {
+      console.log('login component redirectCheck', REDIRECT_URL);
```
Needed?

lunayach commented on src/app/components/utility/loading/loading.component.html:
```diff
@@ -1,7 +1,12 @@
+<div class="loader-container fade" id="loader" *ngIf="authService.isLoader" >
+  <div class="text-med-black center w-300 loader-title">{{authService.loaderTitle}}</div>
+  <div class="loader">
+    <div></div>
```
Can we use *ngFor here?

lunayach replied:
Resolve this :slightly_smiling_face:

lunayach commented on src/styles/styles-port/core/grad-base.scss:
```diff
@@ -0,0 +1,90 @@
+.grad-btn-dark {
+    background: $med-gray;
+    font-weight: $rob-reg;
+    color: #fff;
```
Define the color in `Variables.scss`

lunayach commented on src/styles/styles-port/core/grad-base.scss:
```diff
@@ -0,0 +1,90 @@
+/*font styles*/
+
+.fs-14 {
+    font-size: 14px;
```
For uniformity, can we use `$fs-14` from `variables.scss` instead of `14px`?

lunayach commented on src/styles/styles-port/modules/notification.scss:
```diff
@@ -0,0 +1,67 @@
+        }
+    }
+    label {
+        // color: #ffffff;
```
Remove it.

lunayach reviewed (changes requested):
- UI is not behaving properly. I attached the screenshot above, required changes may be done.
- I see some commented code and log statements in the code, those may be removed.
- I think you are using WebStorm, I recommend you **auto-format** the files before pushing. (`Ctrl+Alt+L`)
- Maybe you could try to have useful comments than "_new css files_", "_Ported From Angular Application_" etc.

sanketbansal commented:
okay, I will make these changes.

sanketbansal committed:
Fixed label overlapping  issue

Removed bottom margin from input field

sanketbansal committed:
Changes in input labels

Made  input labels transit above on focus

sanketbansal committed:
Unnecessary changes removed

Removed Logs and comments

sanketbansal committed:
Added reset password page

Added reset password page for authentication

sanketbansal committed:
Fixed bug in the UI of authentication

Added logic in the UI to transit labels when its value is empty.

sanketbansal committed:
Fixed testing issue

Added services and httpclientmodule, routertestingmodule, formmodule for the test bed of reset-password component

sanketbansal committed:
Fixed unnecessary changes

Removed unused lines of codes, comments and code-format.

lunayach commented:
@sanketbansal 155 file changes! Have all the changes been done for this PR only?

lunayach commented:
@sanketbansal Also, please resolve the PR conflict.

sanketbansal committed:
Moved auth component styles

Styles related to auth component moved from 'styles-port/modules/auth.scss'

sanketbansal committed:
Moved styles to child component of auth page

Styles related to child component moved from auth.component.scss file to corresponding child component scss file

sanketbansal committed:
Fixed bugs in the UI of signup component.

Added color style for password strength label
Removed multiple warming messages for confirm password input

sanketbansal committed:
Fixed UI bug in reset password component

Removed multiple warning message for reset password email input

sanketbansal committed:
Added the loading UI in verify-email component

Added startLoader method to show loading UI while verifying email.

sanketbansal committed:
Fixed lint error in angular.json file

Added comma after a property in angular.json file

sanketbansal committed:
Added reset password confirm component

Reset password confirm component added in auth page

sanketbansal committed:
Fixed bug in reset password component

Added api url to reset password

sanketbansal committed:
Fixed bug in app module file

Fixed json schema of app module file

sanketbansal committed:
Removed images-port folder and styles-port

All the assets moved from images-port folder to images folder
All the styles removed from styles-port folder

sanketbansal committed:
Removed duplicates from base.scss

Duplicates removed from base.scss file

sanketbansal committed:
Added required base styles in the app

Styles-port folder added which contains basic.scss for basic styles

sanketbansal committed:
Removed unused css from main.scss

Duplicates removed from main.scss file

sanketbansal committed:
Removed styles-port main.scss

main.scss file has been removed

sanketbansal committed:
Removed loading styles

Removed loading styles from basic.scss file

sanketbansal committed:
Modified App loading component

Added app loading styles and html

sanketbansal commented:
@lunayach @Shekharrajak  this can be reviewed now.

lunayach commented on src/app/Directives/email.validator.ts:
```diff
@@ -0,0 +1,26 @@
+import {Directive} from '@angular/core';
```
Since you already have `validateEmail` function in `global.service.ts`, no need to create the same function again here. (Also remove the log statement)

sanketbansal replied:
This directive uses validateEmail function in globalService itself.
Done

lunayach commented on src/app/app.module.ts:
```diff
@@ -74,6 +74,10 @@ import { SideBarComponent } from './components/utility/side-bar/side-bar.compone
 import { MatMenuModule } from '@angular/material/menu';
 import { MatIconModule } from '@angular/material/icon';
 
+import {PasswordMismatchValidatorDirective} from './Directives/password.validator';
```
```suggestion
import { PasswordMismatchValidatorDirective } from './Directives/password.validator';
```
Have consistent spacing for imports. (Also at other places.)

lunayach replied:
Resolve this?

sanketbansal replied:
done

lunayach commented on src/app/components/auth/auth.component.scss:
```diff
@@ -1,115 +1,206 @@
+
+.graphic-cont,
+.auth-cont {
+  display: inline-block;
```
Do it like this, `padding: 10vh 40px 40px;`

sanketbansal replied:
Done

lunayach commented on src/app/components/auth/auth.component.ts:
```diff
@@ -10,34 +11,43 @@ import { GlobalService } from '../../services/global.service';
    * @param router  Router Injection.
    * @param route  ActivatedRoute Injection.
    * @param globalService  GlobalService Injection.
+   * @param authService
```
```suggestion
   * @param authService AuthService Injection.
```

sanketbansal replied:
Done

lunayach commented on src/app/components/auth/login/login.component.spec.ts:
```diff
@@ -1,15 +1,16 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import {async, ComponentFixture, TestBed} from '@angular/core/testing';
```
Some of these files changes are only having whitespace (formatting) changes.

lunayach commented on src/app/components/auth/reset-password-confirm/reset-password-confirm.component.html:
```diff
@@ -0,0 +1,71 @@
+        <label for="new_password1" [class.active]="isnewpassword1Focused">New Password (min 8 characters) *</label>
+        <div class="wrn-msg text-highlight" *ngIf="newpassword1.invalid && (newpassword1.dirty || newpassword1.touched || resetconfirmform.submitted)">
+          <p *ngIf="newpassword1.errors.minlength">Password is less than 8 characters.</p>
+          <p *ngIf="newpassword1.errors.required">Password is required.>Password is required.</p>
```
```suggestion
          <p *ngIf="newpassword1.errors.required">Password is required.</p>
```

sanketbansal replied:
Done

lunayach commented on src/app/components/auth/reset-password-confirm/reset-password-confirm.component.ts:
```diff
@@ -0,0 +1,90 @@
+
+  uid: any = '';
+  token: any = '';
+
```
Use camel-casing to define variables i.e. `isNewPaswword1Focused`.

sanketbansal replied:
Done

lunayach commented on src/app/components/auth/reset-password-confirm/reset-password-confirm.component.ts:
```diff
@@ -0,0 +1,90 @@
+              } catch (error) {
+                setTimeout(() => {
+                  this.authService.FormError = 'Something went wrong! Please refresh the page and try again.';
+                  this.globalService.showToast('Error', 'Reset Password UnSuccessfull.Please Try Again!', 5);
```
```suggestion
                  this.globalService.showToast('Error', 'Reset Password UnSuccessful.Please Try Again!', 5);
```

sanketbansal replied:
what here?

lunayach commented on src/app/components/auth/reset-password-confirm/reset-password-confirm.component.ts:
```diff
@@ -0,0 +1,90 @@
+            }
+          },
+
+          () => console.log('RESET-PASSWORD-CONFIRM-FORM-SUBMITTED')
```
Remove this.

lunayach commented on src/app/components/auth/signup/signup.component.html:
```diff
@@ -1,13 +1,98 @@
+                                 (click)="authService.resetForm()">Log In</a></li>
+    <li class="ev-horiz-list"><a routerLink="/auth/signup" class="dark-link" routerLinkActive="active-auth"
+                                 (click)="authService.resetForm()">Sign Up</a></li>
+    <!-- <li class="ev-horiz-list"><a ui-sref="auth.reset-password" class="light-link" ui-sref-active="active-auth" ng-click="auth.resetForm()">Forgot Password</a></li> -->
```
Commented code.

lunayach commented on src/app/components/auth/signup/signup.component.ts:
```diff
@@ -61,43 +60,79 @@ export class SignupComponent implements OnInit, AfterViewInit {
+
+            } catch (error) {
+              setTimeout(() => {
+                this.globalService.showToast('Error', 'Registeration UnSuccessfull.Please Try Again!', 5);
```
```suggestion
                this.globalService.showToast('Error', 'Registration UnSuccessful.Please Try Again!', 5);
```

sanketbansal replied:
what here?

lunayach reviewed (changes requested):
1. I see that the UI of auth components is reflected well. But it's breaking the UI of other components like `home page` etc. Is it due to `styles-port`?
2. Is `Forgot password` feature running smoothly on your side?

sanketbansal committed:
Removed duplicate styles

Removed duplicate scss files from basic.scss

sanketbansal commented:
> 1. I see that the UI of auth components is reflected well. But it's breaking the UI of other components like `home page` etc. Is it due to `styles-port`?
No, actually it do not contains style for older home component.

sanketbansal commented:
> 2. Is `Forgot password` feature running smoothly on your side?
It is sending mail but I don't think I can receive the mail on the local machine or can I?

lunayach commented:
> > 1. Is `Forgot password` feature running smoothly on your side?
> >    It is sending mail but I don't think I can receive the mail on the local machine or can I?

In my case, when I press the `submit` button to send the e-mail, it gets stuck i.e. the loader keeps on looping and even the `Back to Login` button does not work.

lunayach commented:
> > 1. I see that the UI of auth components is reflected well. But it's breaking the UI of other components like `home page` etc. Is it due to `styles-port`?
> >    No, actually it do not contains style for older home component.

So, then we should merge/review the `modified homepage` PR first?

sanketbansal committed:
Resolved issues

Issues resolved in the PR

sanketbansal commented:
> In my case, when I press the `submit` button to send the e-mail, it gets stuck i.e. the loader keeps on looping and even the `Back to Login` button does not work.

It's working now check.

sanketbansal commented:
> So, then we should merge/review the `modified homepage` PR first?
Not necessarily as this PR is independent of modified home page PR.
It's not breaking the UI that I have checked.

lunayach commented:
> > So, then we should merge/review the `modified homepage` PR first?
> > Not necessarily as this PR is independent of modified home page PR.
> > It's not breaking the UI that I have checked.

![one](https://user-images.githubusercontent.com/21227893/61523887-cd94ed00-aa47-11e9-9ed5-7220bffc4a39.png)

lunayach commented:
> > In my case, when I press the `submit` button to send the e-mail, it gets stuck i.e. the loader keeps on looping and even the `Back to Login` button does not work.
> 
> It's working now check.

It does (the frontend part). 
**TODO**: Please check it's email-sending functionality after deployment. (cc: @Sanji515)

sanketbansal committed:
Changes in email validator directive

Removed unused RE expression
Console logs has been removed

RishabhJain2018 commented:
@sanketbansal Please resolve the conflicts here.

sanketbansal commented:
> @sanketbansal Please resolve the conflicts here.

Doing that

sanketbansal commented:
@RishabhJain2018 @lunayach conflicts resolved please review

lunayach commented on src/app/components/auth/signup/signup.component.html:
```diff
@@ -1,13 +1,97 @@
+    <div *ngIf="authService.isFormError" class="text-highlight align-left"> {{authService.FormError}}</div>
+    <div>
+      <p class="text-light-gray">
+        <span class="text-med-black">Already have an account?</span> <a class="highlight-link" routerLink="/auth/login">Log
```
```suggestion
        <span class="text-med-black">Already have an account?</span> &nbsp; <a class="highlight-link" routerLink="/auth/login">Log
```

lunayach commented on src/app/services/auth.service.ts:
```diff
@@ -136,27 +152,57 @@ export class AuthService {
+          error();
+          SELF.globalService.handleApiError(err);
+        },
+        () => console.log('Email Verified')
```
Remove this.

lunayach commented on src/app/components/auth/auth.component.ts:
```diff
@@ -10,34 +11,43 @@ import { GlobalService } from '../../services/global.service';
+    this.authService.resetForm();
+  }
+
+  ngAfterViewInit(): void {
```
Remove this.

lunayach commented on src/app/components/auth/login/login.component.html:
```diff
@@ -1,12 +1,61 @@
-  </div>
+<div class="col s12 m6 offset-m3 left">
+  <ul class="left">
+    <li class="ev-horiz-list"><a routerLink="/auth/login" class="dark-link" routerLinkActive="active-auth"
```
Get `/auth/login` etc. (also check for other such instances in the file) from constant variables in the `.ts` file.

lunayach commented on src/app/components/auth/login/login.component.spec.ts:
```diff
@@ -18,18 +19,18 @@ describe('LoginComponent', () => {
+      schemas: [NO_ERRORS_SCHEMA]
     })
-    .compileComponents();
+      .compileComponents();
```
In the future, here, we should definitely add test cases. After completing the pending PR's, please start opening PR's for adding test cases. (cc: @RishabhJain2018)

lunayach commented on src/app/components/auth/login/login.component.ts:
```diff
@@ -91,25 +70,50 @@ export class LoginComponent implements OnInit, AfterViewInit {
+      return;
+    }
+
+    this.globalService.startLoader('Taking you to EvalAI!');
```
Does this appear? I think I didn't see it when I had tested it.

sanketbansal replied:
yes it was there

sanketbansal committed:
Changes in auth pages

Removed unnecessary changes
Added constant route variable for route in auth components.
Added endpoint paths for reset password and reset password confirm in the endpoint service.

lunayach commented on src/app/components/auth/reset-password-confirm/reset-password-confirm.component.html:
```diff
@@ -0,0 +1,71 @@
+          <button class="waves-effect waves-dark btn ev-btn-dark w-300  grad-btn grad-btn-dark" type="submit" value="Submit">Submit</button>
+        </div>
+        <div class="col s6 align-right right rm-gut">
+          <a routerLink="/auth/login" class="light-link fg-pass">Back to Log In</a>
```
`/auth/login` from `.ts` file. (Check throughout the code.)

sanketbansal replied:
done

Sanji515 commented on src/app/components/auth/login/login.component.html:
```diff
@@ -1,12 +1,61 @@
+  <form name="loginForm" #loginForm='ngForm' (ngSubmit)="userLogin(loginForm.valid)" id="log-in" novalidate>
+    <!-- username -->
+    <div class="input-field align-left">
+      <input type="text" id="name" class="dark-autofill" name="name" (focusin)="isnameFocused = true"
```
@sanketbansal I don't see you using the reusable component for inputs

sanketbansal replied:
yes, because I am using angulars template-driven form.

Sanji515 replied:
but then I think there's no use of `reusable components` if we are not using them.

cc: @RishabhJain2018 @lunayach

lunayach replied:
Thanks @Sanji515 for pointing out this. @sanketbansal can you look at the `input component` in `utility` folder. The component seems to be well generalized. Also, it's `validateInput( )` takes care of things like validating e-mail etc., so we'll not require `email.validator.ts` as well.

sanketbansal replied:
Yeah but it requires a lot of restructuring of the code both in app-input component and auth components. I will create a **seperate PR** for that so that i can use template driven from using app-input component.
Also, I don't think there is any functionality to compare passwords which might need to be added in that.

sanketbansal replied:
Also, There is a difference between validator directive and validateInput() method significantly.
As validator direvtive checks errors at every interaction of user with the input element.

lunayach replied:
Yeah, but that can be taken care while using appropriate events, right?
Still, I feel that the same pattern should be followed in the code. So, if something similar has already been implemented, we should definitely try to use that.
And we can do this re-structuring in a separate PR. @RishabhJain2018, so, should we open an issue for it?

sanketbansal replied:
> Yeah, but that can be taken care while using appropriate events, right?

yeah, but I think its better and effective to use angular features instead of making our own events for that.

sanketbansal replied:
As, validator directive events are being handled by angular internally.

lunayach commented on src/app/components/auth/reset-password-confirm/reset-password-confirm.component.ts:
```diff
@@ -0,0 +1,90 @@
+import { Component, OnInit } from '@angular/core';
```
`Reset-password` and `Reset-password-confirm` component seem to be having many things in common. Do we really need two separate components for two closely-dependent functionalities?

sanketbansal replied:
As both provide different functionality, therefore, I think we should have separate component for them.

lunayach replied:
Tha's true, but I had noticed a lot of similarities in the code for both the components. That's why I brought this. But this is of less-priority, so we can take it in future PR's.

lunayach reviewed (approved):
@RishabhJain2018 LGTM!
There is one thing where @sanketbansal could re-use the `input.component.ts` (after making the required changes) instead of using the Angular template forms. We could have an issue opened for that.

RishabhJain2018 commented:
@sanketbansal Can you please take care of what @lunayach said. Please open up a issue for it.
